contracts/test/integration/IntegrationTestVoltSystemOracle.t.sol

```diff
@@ -0,0 +1,436 @@
+        /// set mint fees to 0 so that the only change that is measured is the
+        /// difference between oracle prices
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        feiPSM.setMintFee(0);
+        usdcPSM.setMintFee(0);
+        vm.stopPrank();
+
+        vm.warp(startTime);
+    }
+
+    function testSetup() public {
+        assertEq(
+            address(oraclePassThrough.scalingPriceOracle()),
+            address(voltSystemOracle)
+        );
+        assertEq(
+            address(existingOraclePassThrough.scalingPriceOracle()),
+            address(scalingPriceOracle)
+        );
+    }
+
+    function testPriceEquivalenceAtTermEnd() public {
+        assertApproxEq(
+            scalingPriceOracle.getCurrentOraclePrice().toInt256(),
+            voltSystemOracle.getCurrentOraclePrice().toInt256(),
+            allowedDeviation
+        );
+        assertEq(
+            voltSystemOracle.oraclePrice(),
+            voltSystemOracle.getCurrentOraclePrice()
+        );
+        assertApproxEq(
+            oraclePassThrough.getCurrentOraclePrice().toInt256(),
+            existingOraclePassThrough.getCurrentOraclePrice().toInt256(),
+            allowedDeviation
+        );
+    }
+
+    /// swap out the old oracle for the new one and ensure the read functions
+    /// give the same value
+    function testMintSwapOraclePassThroughOnPSMs() public {
+        uint256 mintAmount = 100_000e18;
+        uint256 startingAmountOutFei = feiPSM.getMintAmountOut(mintAmount);
+        uint256 startingAmountOutUSDC = usdcPSM.getMintAmountOut(mintAmount);
+
+        vm.warp(startTime - 1 days); /// rewind the clock 1 day so that the timelock execution takes us back to start time
+
+        /// simulate proposal execution so that the next set of assertions can be verified
+        /// with new upgrade in place
+        simulate(
+            getMainnetProposal(),
+            TimelockController(payable(MainnetAddresses.TIMELOCK_CONTROLLER)),
+            MainnetAddresses.GOVERNOR,
+            MainnetAddresses.EOA_1,
+            vm,
+            false
+        );
+
+        uint256 endingAmountOutFei = feiPSM.getMintAmountOut(mintAmount);
+        uint256 endingAmountOutUSDC = usdcPSM.getMintAmountOut(mintAmount);
+
+        assertApproxEq(
+            endingAmountOutFei.toInt256(),
+            startingAmountOutFei.toInt256(),
+            allowedDeviation
+        );
+        assertApproxEq(
+            endingAmountOutUSDC.toInt256(),
+            startingAmountOutUSDC.toInt256(),
+            allowedDeviation
+        );
+    }
+
+    /// swap out the old oracle for the new one and ensure the read functions
+    /// give the same value
+    function testRedeemSwapOraclePassThroughOnPSMs() public {
+        uint256 redeemAmount = 100_000e18;
+        uint256 startingAmountOutFei = feiPSM.getRedeemAmountOut(redeemAmount);
+        uint256 startingAmountOutUSDC = usdcPSM.getRedeemAmountOut(
+            redeemAmount
+        );
```

Comment:
We could probably make this testand it's different variations on both mainnet and arbitrum a fuzz test so we can be sure the behaviour is consistent across all values

Reply:
done on both networks